packages/smb-server/src/server/broker.class.ts

```diff
@@ -0,0 +1,50 @@
+import { randomUUID } from 'crypto';
+import { createServer, Server } from 'net';
+import { Injectable } from 'peque-di';
+import { BrokerSocket, Command, EventService, SocketService } from 'peque-smb-commons/src';
+
+import { MessageCommand, PublishCommand, SubscribeCommand, UnsubscribeCommand, WelcomeCommand } from '../commands';
+
+@Injectable()
+export class Broker {
+  #server: Server;
+  #port = Number(process.env.PORT) ?? 8888;
+  #hostname: string = process.env.HOSTNAME ?? '127.0.0.1';
+
+  constructor(
+    private sockets: SocketService,
+    private events: EventService,
+    private command: Command,
+    private welcomeCommand: WelcomeCommand,
+    private subscribeCommand: SubscribeCommand,
+    private unsubscribeCommand: UnsubscribeCommand,
+    private messageCommand: MessageCommand,
+    private publishCommand: PublishCommand,
+  ) {
+    this.command.init([
+      this.welcomeCommand,
+      this.unsubscribeCommand,
+      this.subscribeCommand,
+      this.messageCommand,
+      this.publishCommand,
+    ]);
+  }
+
+  create(): Promise<void> {
```

Comment:
In this class we might want to have a method to stop listening / close the server.

Reply:
True, indeed the class is still not complete, it will be there, though.